frontend/src/metabase/browse/components/BrowseApp.tsx

```diff
@@ -48,6 +52,62 @@ export const BrowseApp = ({
     }
   }, [tab]);
 
+  const availableModelFilters =
+    PLUGIN_CONTENT_VERIFICATION.availableModelFilters;
+
+  const getInitialModelFilters = () => {
+    return _.reduce(
+      availableModelFilters,
+      (acc, filter, filterName) => {
+        const storedFilterStatus = localStorage.getItem(
+          `browseFilters.${filterName}`,
+        );
+        const shouldFilterBeActive =
+          storedFilterStatus === null
+            ? filter.activeByDefault
+            : storedFilterStatus === "on";
+        return {
+          ...acc,
+          [filterName]: shouldFilterBeActive,
+        };
+      },
+      {},
+    );
+  };
+
+  const [actualModelFilters, setActualModelFilters] =
+    useState<ActualModelFilters>(getInitialModelFilters);
+  const { data: unfilteredModels = [] } = modelsResult;
+
+  const [filteredModels, setFilteredModels] = useState(modelsResult.data);
+
+  useEffect(() => {
+    const filteredModels = _.reduce(
+      actualModelFilters,
+      (acc, shouldFilterBeActive, filterName) =>
+        shouldFilterBeActive
+          ? acc.filter(availableModelFilters[filterName].predicate)
+          : acc,
+      unfilteredModels,
+    );
+    setFilteredModels(filteredModels);
```

Comment:
nit, this would be nice if this logic was extracted in a more testable way. I know the parts are unit tested, but this combination is a little complex and a unit test would be nice of a more encapsulated function

I think it also might be even nicer as a `useMemo`

```ts
const filteredModels = useMemo(() => 
    filterModels(unfilteredModels, actualFilters, availableModelFilters)
, [unfilteredModels, actualFilters]);
```